### ISSUE TITLE
InputStreamAdaptive error in Stan (Omega nightlies / ARMv7 / LibreElec Pi4 and FireTv)

### ISSUE THREAD

**Issue by tekker** [matthuisman/slyguy.addons#546]
Hi Matt, having some problems using Stan with latest Omega Nightlies.  

Trying to access any stream results in error "Unhandled encrypted stream", have reproduced this on a number of different Android devices as well as LibreElec Pi4, all running latest Omega nightlies.  I have also tried installing the latest build of inputstream.adaptive from Github Actions however the error still occurs.  Have tried Android setting of avoid secure decoder to no effect.

Let me know if I you need some more complete logs etc.  Thanks for all your addons mate they are great.

Here is the relevant log on LibreElec Pi4:

```
2023-09-02 22:46:12.462 T:103801    info <general>: VideoPlayer::OpenFile: plugin://plugin.video.stan.au/?_=play&_play=1&play_type=0&profile_id=dee171551faa4a5dab1700f917a23c44&program_id=4183366
2023-09-02 22:46:12.470 T:210697   debug <general>: Thread VideoPlayer start, auto delete: false
2023-09-02 22:46:12.471 T:210697    info <general>: Creating InputStream
2023-09-02 22:46:12.476 T:210689   debug <general>: Loading settings for plugin://plugin.video.stan.au/?_=play&_play=1&play_type=0&profile_id=dee171551faa4a5dab1700f917a23c44&program_id=4183366
2023-09-02 22:46:12.492 T:103801   debug <general>: OnPlayBackStarted: CApplication::OnPlayBackStarted
2023-09-02 22:46:12.530 T:210697   debug <general>: ADDON: Dll Initializing - InputStream Adaptive
2023-09-02 22:46:12.531 T:210697   debug <general>: SECTION:LoadDLL(/storage/.kodi/addons/inputstream.adaptive/inputstream.adaptive.so.21.2.0)
2023-09-02 22:46:12.531 T:210697   debug <general>: Loading: /storage/.kodi/addons/inputstream.adaptive/inputstream.adaptive.so.21.2.0
2023-09-02 22:46:13.782 T:210700   debug <general>: Thread BackgroundLoader start, auto delete: false
2023-09-02 22:46:13.891 T:103801   debug <general>: [threads] name: 'BackgroundLoader' priority: '1'
2023-09-02 22:46:14.105 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Open()
2023-09-02 22:46:14.106 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Property found "inputstream.adaptive.license_data" value: [redacted]
2023-09-02 22:46:14.106 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Property found "inputstream.adaptive.license_key" value: [redacted]
2023-09-02 22:46:14.106 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Property found "inputstream.adaptive.license_type" value: com.widevine.alpha
2023-09-02 22:46:14.132 T:210700   debug <general>: Thread BackgroundLoader 545549762368 terminating
2023-09-02 22:46:14.162 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Property found "inputstream.adaptive.manifest_headers" value: dt-custom-data=eyJ1c2VySWQiOiAiZGVlMTcxNTUxZmFhNGE1ZGFiMTcwMGY5MTdhMjNjNDQiLCJzZXNzaW9uSWQiOiAiZXlKaGJHY2lPaUpJVXpJMU5pSXNJbXRwWkNJNkluQnBhMkZqYUhVaUxDSjBlWEFpT2lKS1YxUWlmUS5leUpwWVhRaU9qRTJPVE0yTlRnM056RXNJbVZ1ZENJNkltaGtJaXdpY0dsa0lqbzBNVGd6TXpZMkxDSm1aV0YwSWpveE1qQTROemd3TmpBNExDSndkWEpqYUdGelpYTWlPakI5LjZXU3NLMTFIblNrUEpCdG9ZdDhzQ29fN3k0aWxJRm9FRl8wSG5aY0NybGciLCJtZXJjaGFudCI6ICJzdGFuIn0%3D&User-Agent=Dalvik/2.1.0%20%28Linux%3B%20U%3B%20Android%209%3B%20SHIELD%20Android%20TV%20Build/PPR1.180610.011%29
2023-09-02 22:46:14.162 T:210697 warning <general>: AddOnLog: inputstream.adaptive: Warning "inputstream.adaptive.manifest_type" property is deprecated and will be removed next Kodi version, the manifest type is now automatically detected.
                                                   If you are using a proxy remember to add the appropriate "content-type" header to the HTTP manifest response
                                                   See Wiki page "How to provide custom manifest/license" to learn more about it.
2023-09-02 22:46:14.162 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Property found "inputstream.adaptive.manifest_type" value: mpd
2023-09-02 22:46:14.163 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Property found "inputstream.adaptive.stream_headers" value: dt-custom-data=eyJ1c2VySWQiOiAiZGVlMTcxNTUxZmFhNGE1ZGFiMTcwMGY5MTdhMjNjNDQiLCJzZXNzaW9uSWQiOiAiZXlKaGJHY2lPaUpJVXpJMU5pSXNJbXRwWkNJNkluQnBhMkZqYUhVaUxDSjBlWEFpT2lKS1YxUWlmUS5leUpwWVhRaU9qRTJPVE0yTlRnM056RXNJbVZ1ZENJNkltaGtJaXdpY0dsa0lqbzBNVGd6TXpZMkxDSm1aV0YwSWpveE1qQTROemd3TmpBNExDSndkWEpqYUdGelpYTWlPakI5LjZXU3NLMTFIblNrUEpCdG9ZdDhzQ29fN3k0aWxJRm9FRl8wSG5aY0NybGciLCJtZXJjaGFudCI6ICJzdGFuIn0%3D&User-Agent=Dalvik/2.1.0%20%28Linux%3B%20U%3B%20Android%209%3B%20SHIELD%20Android%20TV%20Build/PPR1.180610.011%29
2023-09-02 22:46:14.243 T:210697   debug <general>: AddOnLog: inputstream.adaptive: [Repr. chooser] Type: Default
2023-09-02 22:46:14.244 T:210697   debug <general>: AddOnLog: inputstream.adaptive: [Repr. chooser] Configuration
                                                   Resolution max: 0x0
                                                   Resolution max for secure decoder: 0x0
                                                   Bandwidth limits (bit/s): min 0, max 0
                                                   Ignore screen resolution: 0
                                                   Ignore screen resolution change: 0
2023-09-02 22:46:14.244 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Setting NOSECUREDECODER value: 0
2023-09-02 22:46:14.244 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Ignore HDCP status setting value: 1
2023-09-02 22:46:14.244 T:210697    info <general>: AddOnLog: inputstream.adaptive: [Repr. chooser] Resolution set: 1920x1080, max allowed: 3840x2160, Adjust refresh rate: 0
2023-09-02 22:46:14.244 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Searching for decrypters in: /usr/lib/kodi/addons/inputstream.adaptive/
2023-09-02 22:46:14.245 T:210697   error <general>: GetDirectory - Error getting /usr/lib/kodi/addons/inputstream.adaptive/
2023-09-02 22:46:14.245 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Searching for decrypters in:
2023-09-02 22:46:14.245 T:210697   error <general>: GetDirectory - Error getting
2023-09-02 22:46:14.245 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Searching for decrypters in: /storage/.kodi/addons/inputstream.adaptive/
2023-09-02 22:46:14.914 T:103801   debug <general>: CVideoGUIInfo::InitCurrentItem(plugin://plugin.video.stan.au/?_=play&_play=1&play_type=0&profile_id=dee171551faa4a5dab1700f917a23c44&program_id=4183366)
2023-09-02 22:46:14.915 T:103801   debug <general>: CPlayerGUIInfo::InitCurrentItem(plugin://plugin.video.stan.au/?_=play&_play=1&play_type=0&profile_id=dee171551faa4a5dab1700f917a23c44&program_id=4183366)
2023-09-02 22:46:15.015 T:103801   debug <general>: ------ Window Init (DialogBusy.xml) ------
2023-09-02 22:46:17.581 T:103802   debug <UPNP::BuildObject>: Building didl for plain object 'plugin://plugin.video.stan.au/?_=play&_play=1&play_type=0&profile_id=dee171551faa4a5dab1700f917a23c44&program_id=4183366' (encoded value: 'cGx1Z2luOi8vcGx1Z2luLnZpZGVvLnN0YW4uYXUvP189cGxheSZfcGxheT0xJnBsYXlfdHlwZT0wJnByb2ZpbGVfaWQ9ZGVlMTcxNTUxZmFhNGE1ZGFiMTcwMGY5MTdhMjNjNDQmcHJvZ3JhbV9pZD00MTgzMzY2')
2023-09-02 22:46:19.064 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Found decrypter: /storage/.kodi/addons/inputstream.adaptive/libssd_wv.so
2023-09-02 22:46:19.114 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Supported URN: urn:uuid:EDEF8BA9-79D6-4ACE-A3C8-27DCD51D21ED
2023-09-02 22:46:19.257 T:210697   debug <general>: CurlFile::Open - <http://127.0.0.1:52103/https://api.stan.com.au/manifest/v1/dash/androidtv.mpd?url=https%3A%2F%2Fawslive01.stan.video%2Fout%2Fv1%2Fb5abb1d973bd4a75abb8e628a8a27690%2Findex.mpd%3FavailabilityStart%3D1693575366%26clearaudio%3Dtrue%26end%3D1693633319%26live%3Dtrue%26maxQuality%3Dhd%26start%3D1693580407&audioType=all>
2023-09-02 22:46:19.268 T:210697   debug <general>: easy_acquire - Created session to http://127.0.0.1
2023-09-02 22:46:19.424 T:210709   debug <general>: script.module.slyguy - REQUEST IN: https://api.stan.com.au/manifest/v1/dash/androidtv.mpd?url=https%3A%2F%2Fawslive01.stan.video%2Fout%2Fv1%2Fb5abb1d973bd4a75abb8e628a8a27690%2Findex.mpd%3FavailabilityStart%3D1693575366%26clearaudio%3Dtrue%26end%3D1693633319%26live%3Dtrue%26maxQuality%3Dhd%26start%3D1693580407&audioType=all (GET)
2023-09-02 22:46:19.481 T:210709   debug <general>: script.module.slyguy - Session created from proxy data
2023-09-02 22:46:19.781 T:210709   debug <general>: script.module.slyguy - REQUEST OUT: https://api.stan.com.au/manifest/v1/dash/androidtv.mpd?url=https%3A%2F%2Fawslive01.stan.video%2Fout%2Fv1%2Fb5abb1d973bd4a75abb8e628a8a27690%2Findex.mpd%3FavailabilityStart%3D1693575366%26clearaudio%3Dtrue%26end%3D1693633319%26live%3Dtrue%26maxQuality%3Dhd%26start%3D1693580407&audioType=all (GET)
2023-09-02 22:46:20.362 T:210709   debug <general>: script.module.slyguy - SSL Cipher: api.stan.com.au - ('TLS_AES_256_GCM_SHA384', 'TLSv1.3', 256)
2023-09-02 22:46:20.441 T:210709   debug <general>: script.module.slyguy - RESPONSE IN: https://api.stan.com.au/manifest/v1/dash/androidtv.mpd?url=https%3A%2F%2Fawslive01.stan.video%2Fout%2Fv1%2Fb5abb1d973bd4a75abb8e628a8a27690%2Findex.mpd%3FavailabilityStart%3D1693575366%26clearaudio%3Dtrue%26end%3D1693633319%26live%3Dtrue%26maxQuality%3Dhd%26start%3D1693580407&audioType=all (200)
2023-09-02 22:46:20.470 T:210709   debug <general>: script.module.slyguy - Dash Fix: presentationTimeOffset removed
2023-09-02 22:46:20.496 T:210709    info <general>: Skipped 1 duplicate messages..
2023-09-02 22:46:20.496 T:210709   debug <general>: script.module.slyguy - RESPONSE OUT: https://api.stan.com.au/manifest/v1/dash/androidtv.mpd?url=https%3A%2F%2Fawslive01.stan.video%2Fout%2Fv1%2Fb5abb1d973bd4a75abb8e628a8a27690%2Findex.mpd%3FavailabilityStart%3D1693575366%26clearaudio%3Dtrue%26end%3D1693633319%26live%3Dtrue%26maxQuality%3Dhd%26start%3D1693580407&audioType=all (200)
2023-09-02 22:46:20.643 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Download finished: http://127.0.0.1:52103/https://api.stan.com.au/manifest/v1/dash/androidtv.mpd?url=https%3A%2F%2Fawslive01.stan.video%2Fout%2Fv1%2Fb5abb1d973bd4a75abb8e628a8a27690%2Findex.mpd%3FavailabilityStart%3D1693575366%26clearaudio%3Dtrue%26end%3D1693633319%26live%3Dtrue%26maxQuality%3Dhd%26start%3D1693580407&audioType=all|dt-custom-data=eyJ1c2VySWQiOiAiZGVlMTcxNTUxZmFhNGE1ZGFiMTcwMGY5MTdhMjNjNDQiLCJzZXNzaW9uSWQiOiAiZXlKaGJHY2lPaUpJVXpJMU5pSXNJbXRwWkNJNkluQnBhMkZqYUhVaUxDSjBlWEFpT2lKS1YxUWlmUS5leUpwWVhRaU9qRTJPVE0yTlRnM056RXNJbVZ1ZENJNkltaGtJaXdpY0dsa0lqbzBNVGd6TXpZMkxDSm1aV0YwSWpveE1qQTROemd3TmpBNExDSndkWEpqYUdGelpYTWlPakI5LjZXU3NLMTFIblNrUEpCdG9ZdDhzQ29fN3k0aWxJRm9FRl8wSG5aY0NybGciLCJtZXJjaGFudCI6ICJzdGFuIn0%3D&User-Agent=Dalvik/2.1.0%20%28Linux%3B%20U%3B%20Android%209%3B%20SHIELD%20Android%20TV%20Build/PPR1.180610.011%29 (downloaded 4619 byte, speed 3518.00 byte/s)
2023-09-02 22:46:20.738 T:210697    info <general>: AddOnLog: inputstream.adaptive: Manifest successfully parsed (Periods: 1, Streams in first period: 2, Type: VOD)
2023-09-02 22:46:20.739 T:210697   error <general>: AddOnLog: inputstream.adaptive: InitializePeriod: Unhandled encrypted stream.
2023-09-02 22:46:20.739 T:210697   debug <general>: AddOnLog: inputstream.adaptive: [Repr. chooser] Stream selection conditions
                                                   Screen resolution: 1920x1080 (may be limited by settings)
                                                   Initial bandwidth: 3194535 bit/s
2023-09-02 22:46:20.739 T:210697   debug <general>: AddOnLog: inputstream.adaptive: CSession::~CSession()
2023-09-02 22:46:20.836 T:210697   error <general>: CVideoPlayer::OpenInputStream - error opening [plugin://plugin.video.stan.au/?_=play&_play=1&play_type=0&profile_id=dee171551faa4a5dab1700f917a23c44&program_id=4183366]
2023-09-02 22:46:20.932 T:210697    info <general>: CVideoPlayer::OnExit()
2023-09-02 22:46:20.976 T:103801   debug <general>: Flush - flushing renderer
2023-09-02 22:46:21.011 T:210697   debug <general>: AddOnLog: inputstream.adaptive: Close()
2023-09-02 22:46:21.033 T:210697   debug <general>: SECTION:UnloadDll(/storage/.kodi/addons/inputstream.adaptive/inputstream.adaptive.so.21.2.0)
2023-09-02 22:46:21.050 T:210697    info <general>: ADDON: Dll Destroyed - InputStream Adaptive
2023-09-02 22:46:21.117 T:210697   debug <general>: Thread VideoPlayer 545736683328 terminating
```

**Comment by matthuisman** [comment]
have opened an IA issue here:
https://github.com/xbmc/inputstream.adaptive/issues/1372

**Comment by matthuisman** [comment]
fixed in IA 21.2.1
https://github.com/xbmc/inputstream.adaptive/commit/aec6f374e3961145564a23c9edaa66c07164a5ee
